src/libraries/Common/src/System/Security/Cryptography/X509Certificates/Pkcs12LoaderLimits.cs

```diff
@@ -0,0 +1,373 @@
+        /// </remarks>
+        public Pkcs12LoaderLimits()
+        {
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="Pkcs12LoaderLimits"/> class
+        ///   by copying the values from another instance.
+        /// </summary>
+        /// <param name="copyFrom">The instance to copy the values from.</param>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="copyFrom"/> is <see langword="null" />.
+        /// </exception>
+        public Pkcs12LoaderLimits(Pkcs12LoaderLimits copyFrom)
+        {
+#pragma warning disable CA1510
+            if (copyFrom is null)
+                throw new ArgumentNullException(nameof(copyFrom));
+#pragma warning restore CA1510
+
+            // Do not copy _isReadOnly.
+
+            _macIterationLimit = copyFrom._macIterationLimit;
+            _individualKdfIterationLimit = copyFrom._individualKdfIterationLimit;
+            _totalKdfIterationLimit = copyFrom._totalKdfIterationLimit;
+            _maxKeys = copyFrom._maxKeys;
+            _maxCertificates = copyFrom._maxCertificates;
+            _preserveStorageProvider = copyFrom._preserveStorageProvider;
+            _preserveKeyName = copyFrom._preserveKeyName;
+            _preserveCertificateAlias = copyFrom._preserveCertificateAlias;
+            _preserveUnknownAttributes = copyFrom._preserveUnknownAttributes;
+            _ignorePrivateKeys = copyFrom._ignorePrivateKeys;
+            _ignoreEncryptedAuthSafes = copyFrom._ignoreEncryptedAuthSafes;
+        }
+
+        /// <summary>
+        ///   Gets a value indicating whether the instance is read-only.
+        /// </summary>
+        /// <value>
+        ///   <see langword="true" /> if the instance is read-only; otherwise, <see langword="false" />.
+        /// </value>
+        public bool IsReadOnly => _isReadOnly;
+
+        /// <summary>
+        ///   Makes the <see cref="Pkcs12LoaderLimits"/> instance read-only.
+        /// </summary>
+        public void MakeReadOnly()
+        {
+            _isReadOnly = true;
+        }
+
+        private static Pkcs12LoaderLimits MakeReadOnly(Pkcs12LoaderLimits limits)
+        {
+            limits.MakeReadOnly();
+            return limits;
+        }
+
+        private void CheckReadOnly()
+        {
+            if (_isReadOnly)
+            {
+                throw new InvalidOperationException(SR.Cryptography_X509_PKCS12_LimitsReadOnly);
+            }
+        }
+
+        /// <summary>
+        ///   Gets or sets the iteration limit for the MAC calculation.
+        /// </summary>
+        /// <value>The iteration limit for the MAC calculation.</value>
+        public int? MacIterationLimit
+        {
+            get => _macIterationLimit;
+            set
+            {
+                if (value < 0)
+                    throw new ArgumentOutOfRangeException(nameof(value), SR.ArgumentOutOfRange_NeedNonNegNum);
```

Comment:
Earlier an ifdef was used to use ObjectDisposedException.ThrowIf if it's available. I assume we're not doing so here with AOORE just because there are a bunch of them and it'd be abnoxious?

Reply:
That was the reason, yeah; but looking at it now in this file it's just repeating a negative check, so making a helper seems reasonable (it'll get re-used)

Reply:
How "amusing"... I updated every instance except the one this comment was on.  Fixed locally now.